tensorboard/plugins/pr_curve/pr_curve_demo.py

```diff
@@ -0,0 +1,198 @@
+belongs to the class. We use those probabilities to generate PR curves.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+import os.path
+
+from six.moves import xrange  # pylint: disable=redefined-builtin
+import tensorflow as tf
+
+from tensorboard.plugins.pr_curve import summary
+
+FLAGS = tf.flags.FLAGS
+
+tf.flags.DEFINE_string('logdir', '/tmp/pr_curve_demo',
+                       'Directory into which to write TensorBoard data.')
+
+tf.flags.DEFINE_integer('steps', 10,
+                        'Number of steps to generate for each PR curve. '
+                        'Should be less than 40 for reasons in the code.')
+
+
+def start_runs(logdir, steps, run_name, doubly_weigh_blues):
+  """Generate a PR curve with precision and recall evenly weighed.
+  
+  Arguments:
+    logdir: The directory into which to store all the runs' data.
+    steps: The number of steps to run for.
+    run_name: The name of the run.
+    doubly_weigh_blues: Weigh probabilities for whether a color is blue twice
+        as much as default.
+  """
+  tf.reset_default_graph()
+  
+  # Create a normal distribution layer used to generate true color labels.
+  channel_distribution = tf.distributions.Normal(loc=0., scale=42.)
+
+  # Sample the distribution to generate colors. Lets generate different numbers
+  # of each color. The first dimension is the count of examples.
+
+  # Generate reds.
+  number_of_reds = 100
+  true_reds = tf.clip_by_value(
+      tf.concat([
+          255 - tf.abs(channel_distribution.sample([number_of_reds, 1])),
+          tf.abs(channel_distribution.sample([number_of_reds, 2]))], axis=1),
+      0, 255)
+
+  # Generate greens.
+  number_of_greens = 200
+  true_greens = tf.clip_by_value(
+      tf.concat([
+          tf.abs(channel_distribution.sample([number_of_greens, 1])),
+          255 - tf.abs(channel_distribution.sample([number_of_greens, 1])),
+          tf.abs(channel_distribution.sample([number_of_greens, 1]))], axis=1),
+      0, 255)
+
+  # Generate blues.
+  number_of_blues = 150
+  true_blues = tf.clip_by_value(
+      tf.concat([
+          tf.abs(channel_distribution.sample([number_of_blues, 2])),
+          255 - tf.abs(channel_distribution.sample([number_of_blues, 1]))],
+          axis=1),
+      0, 255)
+
+  # Assign each color a vector of 3 booleans based on its true label.
+  labels = tf.concat([
+      tf.tile(tf.constant([[True, False, False]]), (number_of_reds, 1)),
+      tf.tile(tf.constant([[False, True, False]]), (number_of_greens, 1)),
+      tf.tile(tf.constant([[False, False, True]]), (number_of_blues, 1))
+  ], axis=0)
+
+  # We introduce 3 normal distributions. They are used to predict whether a
+  # color falls under a certain class (based on distances from corners of the
+  # color triangle). The distributions vary per color. We have the distributions
+  # narrow over time.
+  iteration = tf.Variable(0.0)
```

Comment:
I'd much prefer that `iteration` be a `tf.placeholder` (`feed_dict={iteration: float(step)}`) than a mutable variable.